pkg/webhooks/webhookpatch.go

```diff
@@ -97,53 +93,49 @@ func PatchCertLoop(injectionWebhookConfigName, webhookName, caBundlePath string,
 				if oldConfig.ResourceVersion != newConfig.ResourceVersion {
 					for i, w := range newConfig.Webhooks {
 						if w.Name == webhookName && !bytes.Equal(newConfig.Webhooks[i].ClientConfig.CABundle, caCertPem) {
-							log.Infof("Detected a change in CABundle, patching MutatingWebhookConfiguration again")
-							shouldPatch <- struct{}{}
+							log.Infof("Detected a change in CA bundle, patching MutatingWebhookConfiguration for %s", newConfig.Name)
+							go doPatchWithRetries(client, revision, newConfig.Name, webhookName, caCertPem)
 							break
 						}
 					}
 				}
 			},
+			AddFunc: func(obj interface{}) {
```

Comment:
Do we need the cabundle check here as well? I think client-go will trigger "add" when istiod starts up, so this would be triggered every time a new istiod starts up. Shouldn't be a huge issue, but just some unneeded api-server calls

Reply:
Yep makes sense, added a check